tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryTests.cs

```diff
@@ -813,6 +814,104 @@ public void MockDirectory_CreateDirectory_ShouldSucceedIfTryingToCreateUNCPathSh
             Assert.IsTrue(fileSystem.Directory.Exists(@"\\server\share\"));
         }
 
+#if FEATURE_CREATE_TEMP_SUBDIRECTORY
+        [Test]
+        public void MockDirectory_CreateTempSubdirectory_ShouldCreateSubdirectoryInTempDirectory()
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory();
+
+            // Assert
+            Assert.IsTrue(fileSystem.Directory.Exists(result.FullName));
+            Assert.IsTrue(result.FullName.StartsWith(Path.GetTempPath()));
+        }
+
+        [Test]
+        public void MockDirectory_CreateTempSubdirectory_ShouldHaveCharactersFromUpperLowerOrDigits()
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory();
+
+            // Assert
+            var name = Path.GetFileName(result.FullName);
+            for (var i = 0; i < name.Length; i++)
+            {
+                var c = name[i];
+                if (!char.IsUpper(c) && !char.IsLower(c) && !char.IsDigit(c))
+                {
+                    Assert.Fail($"Character '{c}' at position {i} of directory {name} is not upper, lower or digit", c, i);
+                }
+            }
+        }
+
+        [Test]
+        public void MockDirectory_CreateTempSubdirectoryWithPrefix_ShouldCreateDirectoryWithGivenPrefixInTempDirectory()
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory("foo-");
+
+            // Assert
+            Assert.IsTrue(fileSystem.Directory.Exists(result.FullName));
+            Assert.IsTrue(Path.GetFileName(result.FullName).StartsWith("foo-"));
+            Assert.IsTrue(result.FullName.StartsWith(Path.GetTempPath()));
+        }
+
+        [Test]
+        public void MockDirectory_CreateTempSubdirectoryWithPrefix_ShouldEndWithCharactersFromUpperLowerOfDigits()
+        {
+            // Arrange
+            var prefix = "foo-";
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory(prefix);
+
+            // Assert
+            var name = Path.GetFileName(result.FullName);
+            for (var i = prefix.Length; i < name.Length; i++)
+            {
+                var c = name[i];
+                if (!char.IsUpper(c) && !char.IsLower(c) && !char.IsDigit(c))
+                {
+                    Assert.Fail($"Character '{c}' at position {i} of directoy {name} is not upper, lower or digit", c, i);
```

Comment:
Typo `directoy` should be `directory`

Reply:
Removed this test, so this typo no longer exists. Test removed as I am now using `Path.GetRandomFileName()` so are now dependent on the .NET standard libraries to determine what is or isn't a valid character for a random file name.